src/plugins/event-plugins/FetchPlugin.ts

```diff
@@ -73,8 +73,12 @@ export class FetchPlugin extends MonkeyPatched<Window, 'fetch'> {
         this.enable();
     }
 
-    private isTracingEnabled = () => {
-        return this.context.config.enableXRay;
+    private isTracingEnabled = (url: string) => {
+        return (
+            this.context.config.enableXRay &&
+            this.config.urlsToTrace.length &&
+            this.config.urlsToTrace.every((re) => re.test(url))
```

Comment:
```suggestion
            this.context.config.enableXRay &&
            this.config.urlsToTrace.every((re) => re.test(url))
```

nit: I think `.every()` evaluates to true when the list is empty

Reply:
Sorry ignore this, I didn't see your intended behavior for empty list

Reply:
```suggestion
            this.context.config.enableXRay &&
            this.config.urlsToTrace.length &&
            this.config.urlsToTrace.some((re) => re.test(url))
```
> the web client will only trace requests that match one of the regular expressions defined in urlsToTrace.

nit: did you mean to use some instead of every?